spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/TypeToSparkType.java

```diff
@@ -130,9 +131,10 @@ public DataType primitive(Type.PrimitiveType primitive) {
       case DECIMAL:
         Types.DecimalType decimal = (Types.DecimalType) primitive;
         return DecimalType$.MODULE$.apply(decimal.precision(), decimal.scale());
+      case UNKNOWN:
+        return NullType$.MODULE$;
       default:
-        throw new UnsupportedOperationException(
-            "Cannot convert unknown type to Spark: " + primitive);
+        throw new UnsupportedOperationException("Type not known to Spark: " + primitive);
```

Comment:
What about `"Cannot convert unsupported type to Spark: "` instead? I like changing the message less, but still removing "unknown" since it is now confusing.